### PR TITLE
[Resolves #302] For v1 - Handled the case in _check_circular_dependencies when there are external dependencies

### LINKED ISSUE #302: Detect Cycles Failing with Sibling Sub-Environments
When I split an environment into two sibling  sub-environments with a dependency between them, the `launch-env` command fails when I try to launch the sub-environment. 

Reproduction steps:
```shell
$ tree
.
├── config
│   ├── bundle
│   │   ├── a
│   │   │   └── x.yaml
│   │   └── b
│   │       └── y.yaml
│   └── config.yaml
└── templates
    └── template.yaml

$ cat config/bundle/a/x.yaml
template_path: template.yaml

parameters:
  SomeInput: "Input"

$ cat config/bundle/b/y.yaml
template_path: template.yaml

parameters:
  SomeInput: !stack_output a/x::SomeOutput

$sceptre launch-env bundle/b
Traceback (most recent call last):
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/bin/sceptre", line 11, in <module>
    sys.exit(cli())
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/sceptre/cli.py", line 96, in decorated
    return func(*args, **kwargs)
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/sceptre/cli.py", line 315, in launch_env
    response = env.launch()
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/sceptre/environment.py", line 127, in launch
    self._check_for_circular_dependencies()
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/sceptre/environment.py", line 351, in _check_for_circular_dependencies
    [stack.name]
  File "/Users/cbosmann/.local/share/virtualenvs/test-wZAZWvZX/lib/python3.6/site-packages/sceptre/helpers.py", line 193, in _detect_cycles
    dependency = available_nodes[dependency_name.split("/")[-1]]
KeyError: 'x'
```

Is this expected behavior? If I remove the cycle detection this works fine.

### PR DESCRIPTION
**Pull request by ankitanallana**
This is similar to an [earlier PR](https://github.com/cloudreach/sceptre/pull/352) raised for the `master` branch for the same issue. When an external dependency is encountered _within_ an environment, it need not be considered for the circular dependency check. 

This has been handled by adding `top_level_path` of an environment in the `_check_circular_dependency()` method and comparing it against the current dependency. If they are not in the same environment being currently executed, we do not consider it for the circular dependency check.

-----------------

* [x] Wrote good commit messages.
* [x] Squashed related commits together after the changes have been approved.
* [x] Commit message starts with `[Resolve #issue-number]` (if a related issue exists).
* [x] Added unit tests.
* [x] ~~Added~~ Modified integration tests (if applicable).
* [x] All unit tests (`make test`) are passing.
* [x] Used the same coding conventions as the rest of the project.
* [x] The new code doesn't generate flake8 (`make lint`) offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

-----------------

P.S. - I am a part of Intuit, Inc and I work with [Omer Azmon](https://github.com/oazmon)

### CODE REVIEW

**Comment by ngfgrant** [comment]
Hey @ankitanallana 

Thanks so much for this PR!

I am hoping to get to this properly in the next couple of days. First pass looks good. 👍

File: sceptre/environment.py [hunk 1]
```diff
@@ -338,13 +338,14 @@ def _check_for_circular_dependencies(self):
         :raises: sceptre.workplan.CircularDependenciesException
         """
         self.logger.debug("Checking for circular dependencies...")
-
+        top_level_path = self.path if self.path and self.path != '.' else ''
```
[the remark this answers is not in the document]
Comment (ankitanallana) [reply sceptre/environment.py]: Sure! I shall do this too!

**Comment by ankitanallana** [comment]
@ngfgrant Hello Niall! I incorporated the change you suggested in both my PRs! Hope it's alright now!

**Comment by akdor1154** [comment]
I was just about to start work on a PR to do exactly this, so huge thanks to @ankitanallana.. any chance this can get in to v1?

**Comment by ngfgrant** [comment]
Closing as won't support in v1. v1 EOL as of tomorrow.

**Pull request closed without merging**